sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/SpannerChangeStreamErrorTest.java

```diff
@@ -0,0 +1,481 @@
+            .build();
+    ResultSetMetadata partitionMetadataResultSetMetadata =
+        ResultSetMetadata.newBuilder()
+            .setRowType(
+                StructType.newBuilder()
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_PARTITION_TOKEN)
+                            .setType(Type.newBuilder().setCode(TypeCode.STRING))
+                            .build())
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_PARENT_TOKENS)
+                            .setType(
+                                Type.newBuilder()
+                                    .setCode(TypeCode.ARRAY)
+                                    .setArrayElementType(
+                                        Type.newBuilder().setCode(TypeCode.STRING)))
+                            .build())
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_START_TIMESTAMP)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_END_TIMESTAMP)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_HEARTBEAT_MILLIS)
+                            .setType(Type.newBuilder().setCode(TypeCode.INT64)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_STATE)
+                            .setType(Type.newBuilder().setCode(TypeCode.STRING)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_WATERMARK)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_CREATED_AT)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_SCHEDULED_AT)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_RUNNING_AT)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .addFields(
+                        Field.newBuilder()
+                            .setName(COLUMN_FINISHED_AT)
+                            .setType(Type.newBuilder().setCode(TypeCode.TIMESTAMP)))
+                    .build())
+            .build();
+    ResultSet getPartitionResultSet =
```

Comment:
nit: Maybe we can create a method to hide this manual creation of the resultSet / metadata so that the test case is smaller?